### ISSUE TITLE
Timestamp override - nested keys

### ISSUE THREAD

**Issue by jaredreich** [Automattic/mongoose#4503]
Perhaps it would be nice to have nested values so one could later also set `created.by`, etc...

``` js
const ItemSchema = new Schema({
   name: { type: String }
}, { timestamps: { createdAt: 'created.at', updatedAt: 'updated.at' } })
```

**Comment by vkarpov15** [comment]
Could be useful, but not a pressing priority. PRs welcome.

**Comment by eizzo** [comment]
@vkarpov15: i'm game to get a pr in motion. what is the project's preferred means for setting values for paths that don't exist? it appears mpath doesn't create, only sets. and lodash isn't a project dependency (decidedly i presume), so `_.set(obj, path, val)` is out of the question.

**Comment by vkarpov15** [comment]
`Document.prototype.set` is the preferred mechanism, it works similarly to _.set()

**Comment by eizzo** [comment]
@vkarpov15: `Document.prototype.set` is fine within the pre-save hook; however, outside of this context one would still need to create paths for various objects throughout `Schema.prototype.setupTimestamp`. Should I be creating a method that mimics `_.set` inside `lib/utils.js`?

**Comment by vkarpov15** [comment]
Hmm and why does mpath not work? Not sure what you mean by "mpath doesn't create, only sets"

**Comment by eizzo** [comment]
```

**Comment by eizzo** [comment]
for example:
```
var obj = {};
mpath.set('a.b.c', 123, obj);
// obj === {}
_.set(obj, 'a.b.c', 123)
// obj === { a: { b: { c: 123 } } }
```

**Comment by vkarpov15** [comment]
If you need it, you can just add a dep on lodash.set https://www.npmjs.com/package/lodash.set

**Comment by vkarpov15** [comment]
This appears to have been fixed, I added a test case covering it in 405bfb3. Works in mongoose >= 4.10.0, looks like this was fixed with #5051 in 4.9.1
